src/bootstrap/lib.rs

```diff
@@ -65,9 +65,15 @@
 //! also check out the `src/bootstrap/README.md` file for more information.
 
 #![deny(warnings)]
+#![feature(associated_consts)]
+#![feature(core_intrinsics)]
```

Comment:
I'm sort of ok with the associated consts, but I'd personally really prefer to remove `core_intrinsics`. I don't think a non-`'static` `TypeId` will exist any time soon due to soundness issues.